src/test/java/org/mule/tools/apikit/output/resources/ResourceGeneratorTest.java

```diff
@@ -0,0 +1,114 @@
+import org.mule.tools.apikit.model.Properties;
+import org.mule.tools.apikit.model.ScaffolderResource;
+import org.mule.tools.apikit.model.ScaffoldingAccessories;
+import org.mule.tools.apikit.model.ScaffoldingConfiguration;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class ResourceGeneratorTest {
+
+  public static final String BASE_PATH = "src/test/resources/org.mule.tools.apikit.output.resources/";
+  public static final String PREFFIX_FULL = "expected-full";
+  public static final String SLASH = "/";
+
+  @Test
+  public void testNoGeneration() {
+    ScaffoldingConfiguration.Builder scaffoldingConfigurationBuilder = ScaffoldingConfiguration.builder();
+    assertNull(ResourcesGenerator.generate(scaffoldingConfigurationBuilder.build()));
+  }
+
+  @Test
+  public void testGenerationYAMLFull() throws IOException {
+    ScaffoldingConfiguration scaffoldingConfiguration = buildScaffoldingConfiguration("yaml/configuration-yaml-full.json", "123");
+    List<ScaffolderResource> generatedResources = ResourcesGenerator.generate(scaffoldingConfiguration);
+    commonAssertResources(generatedResources, 3);
+  }
+
+  @Test
+  public void testGenerationPropertiesFull() throws IOException {
+    ScaffoldingConfiguration scaffoldingConfiguration =
+        buildScaffoldingConfiguration("properties/configuration-properties-full.json", "123");
+    List<ScaffolderResource> generatedResources = ResourcesGenerator.generate(scaffoldingConfiguration);
+    commonAssertResources(generatedResources, 3);
+  }
+
+  private void commonAssertResources(List<ScaffolderResource> generatedResources, int expectedGeneratedResourcesSize)
+      throws IOException {
+    assertEquals(generatedResources.size(), expectedGeneratedResourcesSize);
+    for (ScaffolderResource resource : generatedResources) {
+      String environment = getEnvironment(resource.getName());
+      String extension = getExtension(resource.getName());
+      String expectedFile = getExpectedFile(extension, environment, PREFFIX_FULL);
+      commonAssertResource(resource, environment + "-configuration" + extension, expectedFile);
+    }
+  }
+
+  private String getExpectedFile(String extension, String environment, String preffix) throws IOException {
+    String pathname = BASE_PATH + getFolder(extension) + SLASH + preffix + "-" + environment + extension;
+    return IOUtils.toString(new FileInputStream(new File(pathname)));
+  }
+
+  private String getFolder(String extension) {
+    return extension.replace(".", "");
+  }
+
+  private String getEnvironment(String name) {
+    return getConfiguration(name, 0).split("-")[0];
+  }
+
+  private String getExtension(String name) {
+    return getConfiguration(name, 1).split("-")[0];
```

Comment:
No need of `getConfiguration`. You can just do `name.substring(name.lastIndexOf("."))`